dashboard/app/views/levels/editors/fields/_blockly.html.haml

```diff
@@ -18,8 +18,9 @@
     = render partial: 'levels/editors/fields/checkboxes', locals: {f: f, field_name: :is_k1, description: "Is Pre-Reader level"}
     = render partial: 'levels/editors/fields/checkboxes', locals: {f: f, field_name: :disable_variable_editing, description: "Disable editing of variable names and creation of new variables"}
     = render partial: 'levels/editors/fields/checkboxes', locals: {f: f, field_name: :show_type_hints, description: "Show a small visual hint for what type of block a connection accepts"}
-
-  = render partial: 'levels/editors/fields/checkboxes', locals: {f: f, field_name: :use_modal_function_editor, description: "Use modal function editor"}
+    -# Google Blockly does not currently support the modal function editor
+  - unless (@level.is_a?(Dancelab) || @level.is_a?(Poetry))
```

Comment:
Do we have any Rails-side knowledge of whether a lab uses Google or CDO blockly by default?

Reply:
we do! every Level can respond to `uses_google_blockly?`. the default is defined in level.rb:

https://github.com/code-dot-org/code-dot-org/blob/aabc451911920d8e60d98963995f8416c5cd29f2/dashboard/app/models/levels/level.rb#L608-L610

and individual level types can override it:

https://github.com/code-dot-org/code-dot-org/blob/667d7c8941f9fd1786f2f6c75cf0f43326c61f3a/dashboard/app/models/levels/poetry.rb#L76-L78

Reply:
Nice, good to know! I guess in this case we're not officially on google blockly in dance, so I can see why it's written this way for now. Maybe we can note somewhere that we should update this (maybe already noted in Mike's head :)) when we flip to google blockly in dance to just use that method?

Reply:
Yup, I linked this draft PR in the description above that includes making this change to the dance.rb file: https://github.com/code-dot-org/code-dot-org/pull/48860